test/parallel/test-http-client-close-event.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+const common = require('../common');
+
+// This test ensures that the `'close'` event is emitted after the `'error'`
+// event when a request is made and the socket is closed before we started to
+// receive a response.
+
+const assert = require('assert');
+const http = require('http');
+
+const server = http.createServer(common.mustNotCall());
+
+server.listen(0, common.mustCall(() => {
+  const req = http.get({ port: server.address().port }, common.mustNotCall());
+
+  req.on('error', common.mustCall((err) => {
+    assert.strictEqual(err.constructor, Error);
+    assert.strictEqual(err.message, 'socket hang up');
+    assert.strictEqual(err.code, 'ECONNRESET');
+    req.on('close', common.mustCall(() => server.close()));
```

Comment:
Moving the `req.on('close' ...` line out of the `req.on('error' ...` would make this slightly better. Then just add a boolean that is set in `on('error')` and checked in `on('close')`

Reply:
Done.